bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java

```diff
@@ -70,7 +70,7 @@ protected void receivePacket(DHCPPacket request, @Nullable InetAddress udpRemote
 
         Byte dhcpMessageType = request.getDHCPMessageType();
 
-        if (dhcpMessageType != DHCPPacket.DHCPREQUEST) {
+        if (dhcpMessageType != null && dhcpMessageType != DHCPPacket.DHCPREQUEST) {
```

Comment:
What is the benefit of the null-check? `null` is not equal to `DHCPPacket.DHCPREQUEST` anyway.

Reply:
You're 100% right !